frontend/javascript/README.md

```diff
@@ -0,0 +1,713 @@
+
+Strings look like this:
+```js
+// double quotes
+"this is a string"
+// single quotes
+'this is also a string'
+// backticks (that thing to the left of your 1 key)
+`this is also also a string`
+// you can use a backslash (\) to break long strings into multiple lines
+// this yields "this string spans multiple lines!" (no new line)
+"this string spans \
+multiple lines!"
+// you can use the usual escape sequences
+"newline: \n, tab: \t, carriage return: \r, null: \0, etc"
+// strings are arrays of individual characters. you can access these characters
+// with a subscript ([]). we'll talk more about arrays later!
+// this yields the string '3'
+"012345"[3]
+// + appends strings together. this yields "strings are nice".
+"strings " + "are nice"
+// other types are eagerly converted into strings when appended
+// this yields "this should say false: false"
+"this should say false: " + (false || (false && true))
+// backtick strings allow you to inject any js expression into a string with
+// `${expr}`
+`this should say false: ${false || (false && true)}`
+```
+
+Undefined and null values are kind of special. Undefined is used to indicate that an identifier hasn't been given an explicit value. You'll see undefined values if you try to do something like use a variable that hasn't been assigned to. Null values, on the other hand, are used to indicate an intentional value of 'nothing'. Most of the time, you shouldn't go out of your way to use undefined. Null is maybe a little bit more common.
+```js
+undefined
+null
+```
+
+[^other-primitives]: Okay, technically there's also [`bigint`](https://developer.mozilla.org/en-US/docs/Glossary/BigInt) and [`symbol`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol), but most readers probably don't need to worry about those.
+
+### Variables
+You may from time to time want to store values for later use. You can do this using variables.
+```js
+// declares a new variable `variable0` with initial value `undefined`
+let variable0;
+// variables can be reassigned
+variable0 = "hi";
+// js is dynamically typed, so variables can also be reassigned to values with
+// different types
+variable0 = 0;
+// const variables must be initialized at declaration and cannot be reassigned
+// after declaration
+// const variable1; <- not allowed!
+const variable1 = 1;
+// variable1 = 2; <- not allowed!
+
+// variables are scoped to the block they're in. this means that they're only
+// accessible inside the curly braces ({}) they were declared.
+let outerScope = "outer!";
+// outerScope now accessible
+{
+    // outerScope still accessible
+    let innerScope = "inner!";
+    // innerScope now accessible
+}
+// innerScope no longer accessible
+// outerScope still accessible
+```
```

Comment:
Should we include anything on variable shadowing?

Reply:
Can't speak to this, will leave for Avery or Nivedha.